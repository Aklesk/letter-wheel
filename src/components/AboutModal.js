import React from 'react';
import PropTypes from 'prop-types';
import { Linking, View, Text } from 'react-native';
import Modal from 'react-native-modalbox';
import Hyperlink from 'react-native-hyperlink';

import AboutModalStyle from '../styles/AboutModalStyle';

const AboutModal = ({ isOpen, onClosed }) => (
  <Modal
    backButtonClose
    entry="bottom"
    isOpen={isOpen}
    onClosed={onClosed}
    position="center"
    style={AboutModalStyle.modal}
    swipeToClose
  >
    <View style={AboutModalStyle.container}>
      <Text style={AboutModalStyle.text}>
        {'Target Words is based on the popular newspaper game Target, and inspired by the ' +
        'long-abandoned Target app for Android by Ben Buxton.'}
      </Text>
      <Text style={AboutModalStyle.text}>
        {' '}
      </Text>
      <Hyperlink
        linkStyle={{ color: '#2980b9' }}
        onPress={() => {
          Linking.openURL('https://www.github.com/Fendrian/target-words/');
        }}
      >
        <Text style={AboutModalStyle.text}>
          {'Original copyright Â© 2017-2018 by Wolf Hatch. The source code for this app is released ' +
          'under the MIT license at https://github.com/Fendrian/target-words'}
        </Text>
      </Hyperlink>
      <Text style={AboutModalStyle.text}>
        {' '}
      </Text>
      <Text style={AboutModalStyle.text}>
        {'This work would not be remotely possible without the massive wealth of open-source ' +
        'and public domain resources that it is built upon, a full list of which is available ' +
        'with this project\'s source code on GitHub.'}
      </Text>
    </View>
  </Modal>
);

AboutModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClosed: PropTypes.func.isRequired,
};

export default AboutModal;
