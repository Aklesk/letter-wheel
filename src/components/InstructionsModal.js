import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import Modal from 'react-native-modalbox';

import InstructionsModalStyle from '../styles/InstructionsModalStyle';

const InstructionsModal = ({ isOpen, onClosed }) => (
  <Modal
    backButtonClose
    entry="bottom"
    isOpen={isOpen}
    onClosed={onClosed}
    position="center"
    style={InstructionsModalStyle.modal}
    swipeToClose
  >
    <View style={InstructionsModalStyle.container}>
      <Text style={InstructionsModalStyle.text}>
        {'Target Words is a word-finding game. Select letters from the grid to ' +
        'make words, then click "submit" to see if you found one! Each letter can be used once ' +
        'per word, and every word must contain the center letter.'}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {'Try variations on valid words you find: "Word", "wording", "words", ' +
        '"wordings", and "worded" are all valid, for example, and both American and British spellings ' +
        'are valid.'}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {'For a challenge, try timed mode. Time is added for each correct word, but the ' +
        'clock is relentless!'}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <View>
        <Text style={InstructionsModalStyle.text}>
          {'Speed tips:'}
        </Text>
        <Text style={InstructionsModalStyle.text}>
          {'  • Long-press letters to submit'}
        </Text>
        <Text style={InstructionsModalStyle.text}>
          {'  • Long-press backspace to clear'}
        </Text>
      </View>
      <Text style={InstructionsModalStyle.text}>
        {' '}
      </Text>
      <Text style={InstructionsModalStyle.text}>
        {'Good luck!'}
      </Text>
    </View>
  </Modal>
);

InstructionsModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClosed: PropTypes.func.isRequired,
};

export default InstructionsModal;
