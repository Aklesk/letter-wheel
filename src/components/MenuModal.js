import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TouchableOpacity } from 'react-native';
import { inject, observer } from 'mobx-react';
import Modal from 'react-native-modalbox';

import MenuModalStyle from '../styles/MenuModalStyle';

@inject('store')
@observer
export default class MenuModal extends React.Component {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onClosed: PropTypes.func.isRequired,
    store: PropTypes.shape({
      closeMenuModal: PropTypes.func.isRequired,
      nav: PropTypes.object.isRequired,
      openAboutModal: PropTypes.func.isRequired,
      openInstructionsModal: PropTypes.func.isRequired,
      openMenuModal: PropTypes.func.isRequired,
      scoreGame: PropTypes.func.isRequired,
    }).isRequired,
  }
  render() {
    const {
      closeMenuModal,
      nav,
      openAboutModal,
      openInstructionsModal,
      scoreGame,
    } = this.props.store;
    const {
      container,
      divider,
      line,
      text,
      title,
    } = MenuModalStyle;
    return (
      <Modal
        backButtonClose
        entry="bottom"
        isOpen={this.props.isOpen}
        onClosed={this.props.onClosed}
        position="center"
        style={MenuModalStyle.modal}
        swipeToClose
      >
        <View style={container}>
          <Text style={title}>
            Letter Wheel
          </Text>
          <View style={divider} />
          <TouchableOpacity
            onPress={() => {
              closeMenuModal();
              nav.resetto('New');
            }}
            style={line}
          >
            <Text style={text}>
              New Game
            </Text>
          </TouchableOpacity>
          <View style={divider} />
          <TouchableOpacity
            onPress={() => {
              closeMenuModal();
              scoreGame();
            }}
            style={line}
          >
            <Text style={text}>
              Score This Game
            </Text>
          </TouchableOpacity>
          <View style={divider} />
          <TouchableOpacity
            onPress={openInstructionsModal}
            style={line}
          >
            <Text style={text}>
              Instructions
            </Text>
          </TouchableOpacity>
          <View style={divider} />
          <TouchableOpacity
            onPress={openAboutModal}
            style={line}
          >
            <Text style={text}>
              About Letter Wheel
            </Text>
          </TouchableOpacity>
          <View style={divider} />
        </View>
      </Modal>
    );
  }
}
