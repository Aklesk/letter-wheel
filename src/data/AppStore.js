import { action, computed, observable, runInAction, useStrict } from 'mobx';
import { Alert, Dimensions } from 'react-native';
import { NavigationActions } from 'react-navigation';
import SQLite from 'react-native-sqlite-storage';

useStrict(true);

const alphabet = 'abcdefghijklmnopqrstuvwxyz'.split('');
export default class AppState {
  @observable aboutModal = {};
  @action loadAboutModal = (modal) => {
    this.aboutModal = modal;
  }

  @observable gameModal = {};
  @action loadGameModal = (modal) => {
    this.gameModal = modal;
  }

  @observable instructionsModal = {};
  @action loadInstructionsModal = (modal) => {
    this.instructionsModal = modal;
  }

  @observable letters = '         ';
  @observable loading = false;
  @action setLoading = (bool) => {
    this.loading = bool;
  };
  @observable navigator = {};
  @observable newGameOptions = observable.map({
    timed: false,
    wordRange: {
      min: 10,
      max: 49,
    },
  });
  @action setNewGameOptions = (key, val) => {
    this.newGameOptions.set(key, val);
  }
  @observable orientation = 0;
  @observable scored = false;
  @observable statusText = '';
  @observable timer = -1;
  @action setTimer = (seconds) => {
    this.timer = seconds;
  }
  @observable tried = [];
  @observable width = 0;
  @observable words = [];

  @observable selected = [];
  @action toggleSelected = (letterIndex) => {
    const copy = this.selected.slice();
    const loc = this.selected.indexOf(letterIndex);
    if (loc === -1) {
      this.selected.replace([...copy, letterIndex]);
    } else {
      this.selected.remove(letterIndex);
    }
  }

  // Given a string, this function will randomly rearrange the string then return it.
  shuffle = (string) => {
    const a = string.split('');
    const n = a.length;

    for (let i = n - 1; i > 0; i -= 1) {
      const j = Math.floor(Math.random() * (i + 1));
      const tmp = a[i];
      a[i] = a[j];
      a[j] = tmp;
    }
    return a.join('');
  };

  // Given a word and the SQLite database containing dictionary words, this will return
  // all words which can be made via permutations or sub-permutations of the starting word.
  getPermutatedWords = (word, db) =>
    new Promise((resolve) => {
      db.transaction((tx) => {
        const uniqueLettersInWord = Array.from(new Set(word.split(''))).join('');

        const sqlQuery = `SELECT word FROM words WHERE ${[
          '(length(word) >= 4)',
          '(length(word) <= 9)',
          `(word GLOB '*[${uniqueLettersInWord}]*')`, // INCLUDE words with any of our letters
          ...alphabet.map((letter) => { // EXCLUDE words containing too many of any letter
            const occurrencesInWord = word.split(letter).length - 1;
            return `(word NOT LIKE '${(`%${letter}`).repeat(occurrencesInWord + 1)}%')`;
          }),
        ].join(' AND ')}`;

        new Promise((wordsResolve) => {
          tx.executeSql(sqlQuery, [], (tx1, results2) => {
            wordsResolve(results2.rows.raw().map(row => row.word));
          });
        })
          .then(resolve);
      });
    });

  // Formatting logic for the game word list display
  @computed get triedWordList() {
    if (this.tried.length === 0 && this.scored !== true) {
      return [{ word: 'No words', style: 'neutral' }];
    }
    const triedWords = this.tried.sort().map(word => (
      { word, style: this.words.indexOf(word) !== -1 ? 'correct' : 'incorrect' }
    ));

    // If the game has been scored, display all words
    if (this.scored === true) {
      const notFound = this.words.sort().filter(word => (
        (typeof (this.tried.find(w => (w === word))) === 'undefined')
      )).map(word => (
        ({ word, style: 'neutral' })
      ));
      const yourWords = this.tried.length !== 0 ?
        [
          { word: ' ', style: 'neutral' },
          { word: 'Your words:', style: 'neutral' },
          ...triedWords,
        ]
        :
        [];
      return [
        { word: 'Not found:', style: 'neutral' },
        ...notFound,
        ...yourWords,
      ];
    }

    // ...Otherwise just show words that have been tried
    return triedWords;
  }

  constructor() {
    const { width, height } = Dimensions.get('window');
    Dimensions.addEventListener('change', (data) => {
      runInAction(() => {
        this.orientation = (data.window.width < data.window.height) ? 0 : 1;
        this.width = data.window.width;
      });
    });
    this.orientation = (width < height) ? 0 : 1;
    this.width = Dimensions.get('window').width;

    // Add game score levels
    this.scores = [
      { percent: 0, text: '' },
      { percent: 40, text: 'Good' },
      { percent: 65, text: 'Very Good' },
      { percent: 85, text: 'Excellent' },
      { percent: 100, text: 'Exceptional' },
    ];

    // Open the word database
    const ok = () => {};
    const err = () => {
      Alert.alert('Word database failed to open. Please re-install the app.');
    };
    this.db = SQLite.openDatabase({ name: 'main.db', createFromLocation: 1 }, ok, err);
  }

  nav = {
    goto: (screen) => {
      if (typeof (this.navigator.dispatch) === 'undefined') {
        return false;
      }
      return this.navigator.dispatch({ type: 'Navigation/NAVIGATE', routeName: screen });
    },
    resetto: (screen) => {
      if (typeof (this.navigator.dispatch) === 'undefined') {
        return false;
      }
      return this.navigator.dispatch(NavigationActions.reset({
        index: 0,
        actions: [
          NavigationActions.navigate({ routeName: screen }),
        ],
      }));
    },
  }

  async newGame({
    letters,
    timer,
    tried,
    wordsMax,
    wordsMin,
  }) {
    const onlyWordsContaining = ((letter, words) =>
      words.filter(w => w.indexOf(letter) !== -1)
    );

    let cancel = false;
    let newLetters = '';
    let newWords = [];
    let permutations = [];

    // If we've been passed letters explicitly,
    // use the provided letters and just get word list
    if (typeof (letters) === 'string') {
      newLetters = letters;
      permutations = await this.getPermutatedWords(letters, this.db);
      newWords = onlyWordsContaining(letters.substr(4, 1), permutations);

    // If no letters are provided, find a new word that
    // is suitable for the selected word match range
    // and get word list for it
    } else {
      const queryString = 'SELECT * FROM words WHERE ' +
        'length(word)=9 AND' +
        `${
          alphabet.map((
            char => `(${char} >= ${wordsMin} AND ${char} <= ${wordsMax})`
          ))
            .join(' OR ')
          }` +
        'ORDER BY RANDOM() ' +
        'LIMIT 1';

      await new Promise((resolve) => {
        this.db.transaction((tx) => {
          tx.executeSql(
            queryString,
            [],
            async (x, { rows }) => {
              const row = rows.item(0);

              if (!row) {
                Alert.alert('No suitable words found. Please expand search parameters.');
                cancel = true;
              } else {
                const centerLetter = this.shuffle(alphabet.join(''))
                  .split('')
                  .find(key => row[key] >= wordsMin && row[key] <= wordsMax);

                const jumble = this.shuffle(row.word).replace(centerLetter, '');
                newLetters = `${jumble.substr(0, 4)}${centerLetter}${jumble.substr(4)}`;

                permutations = await this.getPermutatedWords(newLetters, this.db);
                newWords = onlyWordsContaining(centerLetter, permutations);
              }
              resolve();
            },
          );
        });
      });
    }

    // Load up the data store with the results
    if (!cancel) {
      runInAction(() => {
        this.letters = newLetters;
        this.words.replace(newWords);
        this.tried.replace(Array.isArray(tried) ? tried : []);
        this.selected.replace([]);
        this.scored = false;
        this.statusText = 'Welcome!';

        if (typeof (timer) === 'number') {
          this.timer = timer === -1 ? ((newWords.length - 1) * 4) : timer;
        } else {
          this.timer = -1;
        }
        console.log(newWords);
      });
    }
  }

  @action submitWord = () => {
    // If game has already been scored, fail
    if (this.scored === true) {
      this.setStatus('Game already scored');
      return false;
    }

    // If word too short, fail
    if (this.selected.length < 4) {
      this.setStatus('Too short');
      return false;
    }

    const word = this.selected.map(i => (
      this.letters.substr(i, 1)
    )).join('');

    // If the middle letter isn't selected, fail
    if (word.indexOf(this.letters.substr(4, 1)) === -1) {
      this.setStatus('Missing middle letter');
      return false;
    }

    // If word already guessed, fail
    if (this.tried.find(w => (w === word))) {
      this.setStatus('Already tried');
      return null;
    }

    // If the word is correct, report to user and add time if relevant
    if (this.words.indexOf(word) !== -1) {
      this.tried.push(word);
      this.selected.replace([]);
      if (this.timer > -1) {
        const addTime = (word.length * 5);
        this.setStatus(`Nice! +${addTime} seconds.`);
        this.timer = (this.timer + addTime);
      } else {
        this.setStatus('Nice!');
      }
      return true;
    }

    // Finally, just fail
    this.tried.push(word);
    this.selected.replace([]);
    this.setStatus('Unrecognized word.');
    return false;
  }

  @action setStatus = (message) => {
    this.statusText = message;
    setTimeout(action(() => {
      if (this.statusText === message) {
        if (this.scored === false) {
          this.statusText = '';
        } else {
          this.statusText = this.getScore();
        }
      }
    }), 3000);
  }

  getScore = () => {
    const correct = this.tried.filter(tryEntry => (
      this.words.indexOf(tryEntry) !== -1
    )).length;
    const adjustedScores = this.scores.map(scoreObj => (
      {
        ...scoreObj,
        numWords: (Math.floor((scoreObj.percent / 100) * this.words.length)),
      }
    ));
    const currentScore = adjustedScores
      .filter(a => (a.numWords <= correct))
      .sort((a, b) => b.numWords - a.numWords)[0];
    const nextScore = adjustedScores
      .filter(a => (a.numWords > correct))
      .sort((a, b) => a.numWords - b.numWords)[0];
    return ({
      text: currentScore.text,
      toNext: (typeof (nextScore) !== 'undefined') ?
        (nextScore.numWords - correct)
        :
        0,
    });
  }

  @action scoreGame = () => {
    this.scored = true;
    this.statusText = 'Game scored';
  }
}
