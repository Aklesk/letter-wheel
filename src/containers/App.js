import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { inject, observer } from 'mobx-react';
import Spinner from 'react-native-loading-spinner-overlay';

import MenuModal from '../components/MenuModal';
import MenuModalStyle from '../styles/MenuModalStyle';
import AboutModal from '../components/AboutModal';
import AboutModalStyle from '../styles/AboutModalStyle';

// Until react-native-modalbox updates, use this to stop backAndroid warnings.
import Modal from '../react-native-modalbox';

@inject('appStore') @observer
class App extends React.Component {
  static propTypes = {
    appStore: PropTypes.shape({
      nav: PropTypes.object.isRequired,
    }).isRequired,
    children: PropTypes.element.isRequired,
  }
  render() {
    const { appStore, children } = this.props;
    return (
      <View style={{ flex: 1 }}>
        <Spinner visible={appStore.loading} textContent={'Loading...'} textStyle={{ color: '#FFF' }} />
        {children}
        <Modal
          backButtonClose
          entry={'bottom'}
          position={'center'}
          style={MenuModalStyle.modal}
          ref={(ref) => { appStore.gameModal = ref; }}
          swipeToClose
        >
          <MenuModal />
        </Modal>
        <Modal
          backButtonClose
          entry={'bottom'}
          position={'center'}
          style={AboutModalStyle.modal}
          ref={(ref) => { appStore.aboutModal = ref; }}
          swipeToClose
        >
          <AboutModal />
        </Modal>
      </View>
    );
  }
}

export default App;
