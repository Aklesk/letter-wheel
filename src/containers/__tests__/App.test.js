import React from 'react';
import { mount } from 'enzyme';
import Spinner from 'react-native-loading-spinner-overlay';
import { runInAction } from 'mobx';

import AppStore from '../../data/store';
import App from '../App';
import MenuModal from '../../components/MenuModal';
import AboutModal from '../../components/AboutModal';
import InstructionsModal from '../../components/InstructionsModal';

jest.mock('react-native-loading-spinner-overlay', () => {
  require('react'); // eslint-disable-line global-require
  const { View } = require('react-native'); // eslint-disable-line global-require
  return props => (
    <View {...props}>
      {'7A9FEEAE-7F90-4135-B5B2-356DB9E822CB'}
    </View>
  );
});
jest.mock('../../components/MenuModal', () => {
  require('react'); // eslint-disable-line global-require
  const { View } = require('react-native'); // eslint-disable-line global-require
  return props => (
    <View {...props}>
      {'D6BA4E9F-2C03-43BE-A2CB-9005355CE3C0'}
    </View>
  );
});
jest.mock('../../components/AboutModal', () => {
  require('react'); // eslint-disable-line global-require
  const { View } = require('react-native'); // eslint-disable-line global-require
  return props => (
    <View {...props}>
      {'2B8A071D-EAE0-4755-B3DC-1DD7AA0B4725'}
    </View>
  );
});
jest.mock('../../components/InstructionsModal', () => {
  require('react'); // eslint-disable-line global-require
  const { View } = require('react-native'); // eslint-disable-line global-require
  return props => (
    <View {...props}>
      {'8E4CFDDB-A054-462D-B718-C182E846D653'}
    </View>
  );
});

describe('Main App component', () => {
  let store;

  beforeEach(() => {
    store = new AppStore();
  });

  afterEach(() => {
    jest.clearAllMocks();
    store = undefined;
  });

  it('matches the snapshot', () => {
    const render = mount((
      <App store={store}>
        C3D066C6-EFB0-4E46-AC72-6E2558B34F1B
      </App>
    ));
    expect(render).toMatchSnapshot();
  });

  it('exports the App component', () => {
    const render = mount((
      <App store={store}>
        616DC9D6-CC2B-4D7A-8B54-B610A2422593
      </App>
    ));
    expect(render.find(App)).toHaveLength(1);
  });

  it('renders a loading spinner', () => {
    const render = mount((
      <App store={store}>
        5F82BC43-6BF3-41F2-A215-CB9F44F208F1
      </App>
    ));
    expect(render.find(Spinner)).toHaveLength(1);
  });

  it('renders the Menu modal', () => {
    const render = mount((
      <App store={store}>
        289FAD8B-45FB-496A-9FEB-AD8DEDD75F25
      </App>
    ));
    expect(render.find(MenuModal)).toHaveLength(1);
    expect(render.find(MenuModal).props().onClosed).toEqual(store.closeMenuModal);
    expect(render.find(MenuModal).props().isOpen).toBeFalsy();
    runInAction(() => { store.isMenuModalOpen = true; });
    render.update();
    expect(render.find(MenuModal).props().isOpen).toBeTruthy();
    runInAction(() => { store.isMenuModalOpen = false; });
    render.update();
    expect(render.find(MenuModal).props().isOpen).toBeFalsy();
  });

  it('renders the About modal', () => {
    const render = mount((
      <App store={store}>
        43032E29-AECF-4389-9D2F-665CBFC13DE9
      </App>
    ));
    expect(render.find(AboutModal)).toHaveLength(1);
    expect(render.find(AboutModal).props().onClosed).toEqual(store.closeAboutModal);
    expect(render.find(AboutModal).props().isOpen).toBeFalsy();
    runInAction(() => { store.isAboutModalOpen = true; });
    render.update();
    expect(render.find(AboutModal).props().isOpen).toBeTruthy();
    runInAction(() => { store.isAboutModalOpen = false; });
    render.update();
    expect(render.find(AboutModal).props().isOpen).toBeFalsy();
  });

  it('renders the Instructions modal', () => {
    const render = mount((
      <App store={store}>
        B9B8DC2D-99D7-4A0F-B7FD-A1CD6A74978D
      </App>
    ));
    expect(render.find(InstructionsModal)).toHaveLength(1);
    expect(render.find(InstructionsModal).props().onClosed).toEqual(store.closeInstructionsModal);
    expect(render.find(InstructionsModal).props().isOpen).toBeFalsy();
    runInAction(() => { store.isInstructionsModalOpen = true; });
    render.update();
    expect(render.find(InstructionsModal).props().isOpen).toBeTruthy();
    runInAction(() => { store.isInstructionsModalOpen = false; });
    render.update();
    expect(render.find(InstructionsModal).props().isOpen).toBeFalsy();
  });

  it('returns on render', () => {
    const render = mount((
      <App store={store}>
        B9B8DC2D-99D7-4A0F-B7FD-A1CD6A74978D
      </App>
    ));
    expect(render.find(InstructionsModal)).toHaveLength(1);
    expect(render.find(InstructionsModal).props().onClosed).toEqual(store.closeInstructionsModal);
    expect(render.find(InstructionsModal).props().isOpen).toBeFalsy();
    runInAction(() => { store.isInstructionsModalOpen = true; });
    render.update();
    expect(render.find(InstructionsModal).props().isOpen).toBeTruthy();
    runInAction(() => { store.isInstructionsModalOpen = false; });
    render.update();
    expect(render.find(InstructionsModal).props().isOpen).toBeFalsy();
  });
});
