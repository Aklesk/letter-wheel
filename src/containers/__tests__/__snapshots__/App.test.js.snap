// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App component matches the snapshot 1`] = `
<inject-App-with-store
  store={
    store {
      "closeAllModals": [Function],
      "db": Object {
        "transaction": [MockFunction],
      },
      "getPermutatedWords": [Function],
      "isAboutModalOpen": false,
      "isInstructionsModalOpen": false,
      "isMenuModalOpen": false,
      "letters": "         ",
      "loading": false,
      "nav": Object {
        "get": [Function],
        "goto": [Function],
        "resetto": [Function],
      },
      "navigator": Object {},
      "newGameOptions": Object {
        "timed": false,
        "wordRange": Object {
          "max": 49,
          "min": 10,
        },
      },
      "orientation": 0,
      "scored": false,
      "scores": Array [
        Object {
          "percent": 100,
        },
        Object {
          "percent": 85,
        },
        Object {
          "percent": 65,
        },
        Object {
          "percent": 40,
        },
        Object {
          "percent": 0,
        },
      ],
      "selected": Array [],
      "shuffle": [Function],
      "statusText": "",
      "timer": -1,
      "tried": Object {},
      "width": 320,
      "words": Object {},
    }
  }
>
  <App
    store={
      store {
        "closeAllModals": [Function],
        "db": Object {
          "transaction": [MockFunction],
        },
        "getPermutatedWords": [Function],
        "isAboutModalOpen": false,
        "isInstructionsModalOpen": false,
        "isMenuModalOpen": false,
        "letters": "         ",
        "loading": false,
        "nav": Object {
          "get": [Function],
          "goto": [Function],
          "resetto": [Function],
        },
        "navigator": Object {},
        "newGameOptions": Object {
          "timed": false,
          "wordRange": Object {
            "max": 49,
            "min": 10,
          },
        },
        "orientation": 0,
        "scored": false,
        "scores": Array [
          Object {
            "percent": 100,
          },
          Object {
            "percent": 85,
          },
          Object {
            "percent": 65,
          },
          Object {
            "percent": 40,
          },
          Object {
            "percent": 0,
          },
        ],
        "selected": Array [],
        "shuffle": [Function],
        "statusText": "",
        "timer": -1,
        "tried": Object {},
        "width": 320,
        "words": Object {},
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <react-native-mock>
        <StatusBar
          backgroundColor="#00000055"
          translucent={true}
        >
          <react-native-mock />
        </StatusBar>
        <Component
          textContent="Loading..."
          textStyle={
            Object {
              "color": "#FFF",
            }
          }
          visible={false}
        >
          <View
            textContent="Loading..."
            textStyle={
              Object {
                "color": "#FFF",
              }
            }
          >
            <react-native-mock>
              7A9FEEAE-7F90-4135-B5B2-356DB9E822CB
            </react-native-mock>
          </View>
        </Component>
        <Text>
          <react-native-mock>
            C3D066C6-EFB0-4E46-AC72-6E2558B34F1B
          </react-native-mock>
        </Text>
        <Component
          isOpen={false}
          onClosed={[Function]}
        >
          <View
            isOpen={false}
            onClosed={[Function]}
          >
            <react-native-mock>
              D6BA4E9F-2C03-43BE-A2CB-9005355CE3C0
            </react-native-mock>
          </View>
        </Component>
        <Component
          isOpen={false}
          onClosed={[Function]}
        >
          <View
            isOpen={false}
            onClosed={[Function]}
          >
            <react-native-mock>
              2B8A071D-EAE0-4755-B3DC-1DD7AA0B4725
            </react-native-mock>
          </View>
        </Component>
        <Component
          isOpen={false}
          onClosed={[Function]}
        >
          <View
            isOpen={false}
            onClosed={[Function]}
          >
            <react-native-mock>
              8E4CFDDB-A054-462D-B718-C182E846D653
            </react-native-mock>
          </View>
        </Component>
      </react-native-mock>
    </View>
  </App>
</inject-App-with-store>
`;
